Documentation/proposals/202304-silences.md

```diff
@@ -0,0 +1,112 @@
+# Silence CRD
+
+* Owners:
+  * [mcbenjemaa](https://github.com/mcbenjemaa)
+* Related Tickets:
+  * [#5452](https://github.com/prometheus-operator/prometheus-operator/issues/5452)
+  * [#2398](https://github.com/prometheus-operator/prometheus-operator/issues/2398)
+* Other docs:
+  * n/a
+
+This document describes the creation of `Silence` Custom Resource Definition that defines Alertmanager silences
+configurations in the Kubernetes way.
+
+# Why
+
+Prometheus-operator doesn't have a way to automate the management of Alertmanager silences. Users have either been using internal scripts
+or an additional operator that does the job e.g. [silence-operator](https://github.com/giantswarm/silence-operator).
+
+* Users using CI/CD jobs to manage silences have reported that this is cumbersome, since they need to either expose
+  Alertmanager via an ingress or using a service account to connect to the cluster, and this is even more complicated
+  in a multi-cluster setup.
```

Comment:
> Another use case where this would be useful is when enabling AlertmanagerConfig custom resources to offer self-service notifications.

what does this mean?

Reply:
In GS, our original use case is that we needed to deploy the same silences in multiple clusters from a single source of truth.

Deploying flux is definitely easier on multiple clusters than deploying some internal tooling to sync silences and this is all done from within the cluster so using the alertmanager is enough.

In a CI way (unless you use something like tekton for sure) you would need to have your CI aware about all the different alertmanager URIs, but also have it on every silence which can be expensive depending on the number of clusters you have. Also, if the ci cannot access an alertmanager for some reason, you would need to rerun the CI again